### PR TITLE
docs(pharos-site): add how to write documentation section and contributing section

### PR DESCRIPTION
**Pull request by Lham42**
**This change:** (check at least one)

- [x] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [x] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [x] Changeset added?

**What does this change address?**
This change helps encourage more people to contribute to the design system and documentation.

**How does this change work?**
A page was added detailing how to write documentation. Further, a dropdown was added to the sidenav for contributing related content.

**Additional context**
Accessibility section has yet to be added.

### CODE REVIEW

File: packages/pharos-site/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<h1 align="center">
+  Publishing documentation
+</h1>
+
+## MDX
+
+Documentation pages on the Pharos site are written using MDX — markdown files that allow you to integrate JSX easily. This combines the ease in editing long-form content with being able to import components which can elevate the content. These file names should be kebab-cased and named to match the component and what will be displayed in the side navigation bar.
+
+## PHaros components
```
Comment (Niznikr) [inline packages/pharos-site/README.md:9]:
```suggestion
## Pharos components
```

File: packages/pharos-site/src/components/Sidenav.tsx [hunk 2]
```diff
@@ -76,6 +76,9 @@ const Sidenav: FC = () => {
           >
             Help
           </PharosSidenavLink>
+          <PharosSidenavMenu label="Contributing" expanded={isExpanded('contributing')}>
+            {['Contributing documentation'].map(createSidenavLink.bind(this, 'contributing'))}
```
Comment (Lham42) [inline packages/pharos-site/src/components/Sidenav.tsx:80]: I kinda don't like how on the sidenav the title takes up two lines. Thoughts?

File: packages/pharos-site/src/pages/contributing/contributing-documentation.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    </PageSection>
+    <PageSection
+      title="Content guidelines (If applicable)"
+      subSectionLevel={2}
+      description="If a component has content, please include guidelines for how the content should be written for the needs of the component (e.g., how labels should be written)."
+    ></PageSection>
+    <PageSection
+      title="Variants (If applicable)"
+      subSectionLevel={2}
+      description="If a component has variants, display an example of the Pharos variant along with their title, purposes, and use cases(e.g. Error, Warning, Info, and Success Alerts). "
```
Comment (daneah) [inline packages/pharos-site/src/pages/contributing/contributing-documentation.mdx:91]:
```suggestion
      description="If a component has variants, display an example of the Pharos variant along with their title, purposes, and use cases (e.g. Error, Warning, Info, and Success Alerts). "
```

File: packages/pharos-site/src/components/Sidenav.tsx (hunk 2 above)
Comment (Niznikr) [reply packages/pharos-site/src/components/Sidenav.tsx:80]: Think the title for the item can just be `Documentation`

File: packages/pharos-site/src/pages/contributing/documentation.mdx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  description="When a new component or feature is added, supporting documentation should be written to communicate and provide guidelines for its proper usage. 
+Additionally, if changes are made to existing components that affect their usage or functionality, the documentation should reflect these changes."
+></PageSection>
+<PageSection title="Writing documentation">
+  <PageSection title="Getting started" subSectionLevel={1}>
+    <p>
+      To get started, use our
+      <PharosLink href="https://docs.google.com/document/d/1GDFM4iQsta8MjErF7hEYkFaNMoQNKzEzpM-RWG8JcwM/edit">
+        {' '}
+        documentation template{' '}
```
Comment (Lham42) [inline packages/pharos-site/src/pages/contributing/documentation.mdx:26]: ill get rid of these

Comment (Niznikr) [reply packages/pharos-site/src/pages/contributing/documentation.mdx:26]: Think you want those for the spaces between the link and text

File: packages/pharos-site/src/pages/contributing/documentation.mdx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+></PageSection>
+<PageSection
+  title="When documentation is needed"
+  description="When a new component or feature is added, supporting documentation should be written to communicate and provide guidelines for its proper usage. 
+Additionally, if changes are made to existing components that affect their usage or functionality, the documentation should reflect these changes."
+></PageSection>
+<PageSection title="Writing documentation">
+  <PageSection title="Getting started" subSectionLevel={1}>
+    <p>
+      To get started, use our
```
Comment (daneah) [inline packages/pharos-site/src/pages/contributing/documentation.mdx:23]: This "our" feels awkward:

```suggestion
      To get started, use the
```

File: packages/pharos-site/src/pages/contributing/documentation.mdx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    ></PageSection>
+    <PageSection
+      title="Deck"
+      subSectionLevel={2}
+      description="Tweet-length description of the component and its purpose."
+    ></PageSection>
+    <PageSection
+      title="Usage"
+      subSectionLevel={2}
+      description="Explain the implementation of the element (e.g. orientation, placement, use cases). Indicate any behaviour that a user would need to know that should happen when using this component in their UI."
```
Comment (daneah) [inline packages/pharos-site/src/pages/contributing/documentation.mdx:69]:
```suggestion
      description="Explain the implementation of the element (e.g. orientation, placement, use cases). Indicate any behavior that a user would need to know that should happen when using this component in their UI."
```

File: packages/pharos-site/src/pages/contributing/documentation.mdx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    </p>
+    <PageSection
+      title="Concise"
+      subSectionLevel={2}
+      description="The way we write is simple and direct, and for all reading levels. It should be understandable by anyone regardless of their culture, language, or ability. We aim to provide the right amount of context so our users know how to use Pharos components or elements successfully."
+    ></PageSection>
+    <PageSection
+      title="Consistent"
+      subSectionLevel={2}
+      description="We use the same template for each of our components to create predictability in understanding their purpose, standards, and variations. We also aim to use consistent language in all documentation so users experience a seamless transition as they make their way through the site."
```
Comment (daneah) [inline packages/pharos-site/src/pages/contributing/documentation.mdx:47]:
```suggestion
      description="We use the same template for each of the components to create predictability in understanding their purpose, standards, and variations. We also aim to use consistent language in all documentation so users experience a seamless transition as they make their way through the site."
```

File: packages/pharos-site/src/pages/contributing/documentation.mdx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+      To get started, use our
+      <PharosLink href="https://docs.google.com/document/d/1GDFM4iQsta8MjErF7hEYkFaNMoQNKzEzpM-RWG8JcwM/edit">
+        {' '}
+        documentation template{' '}
+      </PharosLink>&nbsp; to ensure all needed sections are included and consistent.
+    </p>
+  </PageSection>
+  <PageSection title="Writing guidelines" subSectionLevel={1}>
+    <p>
+      Please refer to our
```
Comment (daneah) [inline packages/pharos-site/src/pages/contributing/documentation.mdx:32]:
```suggestion
      Please refer to the
```

File: packages/pharos-site/src/pages/contributing/documentation.mdx (hunk 6 above)
Comment (Lham42) [reply packages/pharos-site/src/pages/contributing/documentation.mdx]: 🇨🇦 spelling >>> 
but will change

**Review by daneah — APPROVED** [review]

**Review by Niznikr — APPROVED** [review]

**Pull request merged**